cupy/core/raw.pyx

```diff
@@ -205,16 +209,21 @@ cdef class RawKernel:
 def _get_raw_kernel(str code, str path, str name, tuple options=(),
                     str backend='nvrtc',
                     bint translate_cucomplex=False,
-                    bint enable_cooperative_groups=False):
+                    bint enable_cooperative_groups=False,
+                    tuple specializations=None):
     cdef Module mod
     cdef Function ker
     assert (code is None) != (path is None)
     mod = _get_raw_module(code, path, options, backend,
-                          translate_cucomplex, enable_cooperative_groups)
+                          translate_cucomplex, enable_cooperative_groups,
+                          specializations)
     ker = mod.get_function(name)
     return ker
 
 
+cdef object _cpp_template_ker_name = re.compile(r'\s*'.join(['<', '.+', '>']))
```

Comment:
Would be nice to have someone proficient in regex to take a look (I know very little 😅). The goal is to tell if a string contains a pair of `<` and `>`, such as `func<int>` and `func2<complex<double>>`.

Reply:
This implementation cannot support [this use case](https://github.com/cupy/cupy/pull/3319#issuecomment-623054903).

Reply:
That's OK because we then would not look up the (mangled) name from the specialization dict. `test_raw.py` passed locally, which does contain that case.

Reply:
Sorry I get it now. So I should simply just try retrieving the mangled name?

Reply:
See 1b5de18.